### PR TITLE
chore(relational_tests): refactor relational table unit tests

### PR DESCRIPTION
**Pull request by wcy-fdu**
## What's changed and what's your intention?

Since there are `StateTable`, `MemTable` and `CellBasedTable` in relational layer, this PR moves the related unit tests into one file called test_relational_test.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

### CODE REVIEW

File: src/storage/Cargo.toml [hunk 1]
```diff
@@ -30,6 +30,7 @@ lazy_static = "1"
 libc = "0.2"
 log = "0.4"
 lz4 = "1.23.1"
+madsim = "0.2.0-alpha.1"
```
Comment (BowenXiao1999) [inline src/storage/Cargo.toml:33]: already used madsim @wangrunji0408 ?

**Review by BowenXiao1999 — APPROVED** [review]
LGTM

File: src/storage/Cargo.toml (hunk 1 above)
Comment (wangrunji0408) [reply src/storage/Cargo.toml:33]: I guess not yet in this PR 😂

Comment (wcy-fdu) [reply src/storage/Cargo.toml:33]: Ok, I will remove it later.

**Pull request merged**